### PR TITLE
sync_templates_and_ci: use rsync

### PR DESCRIPTION
**Pull request by jawshooah**
`rsync` allows us to delete files present at the destination but not the source.

Also split arguments into multiple lines for readability/maintainability.

### CODE REVIEW

File: developer/bin/sync_templates_and_ci [hunk 1]
```diff
@@ -31,7 +31,11 @@ function copy_templates_and_ci {
   local repo_name
 
   repo_name="${1}"
-  cp -R "${main_repo_dir}/.github" "${main_repo_dir}/.travis.yml" "${main_repo_dir}/ci" '.'
+  rsync --archive --delete \
+    "${main_repo_dir}/.github" \
```
Comment (jawshooah) [inline developer/bin/sync_templates_and_ci]: Could also shorten this list of sources to `"${main_repo_dir}"/{.github,.travis.yml,ci}` but figured splitting them up would be more readable.

**Comment by vitorgalvao** [comment]
Was just looking into this. It appears GNU `cp` could delete the files not present on the source but BSD `cp` cannot.

File: developer/bin/sync_templates_and_ci (hunk 1 above)
Comment (vitorgalvao) [inline developer/bin/sync_templates_and_ci]: I’d argue keeping them in a single line is readable. Ditto for `"${main_repo_dir}"/{.github,.travis.yml,ci}`, it’s simple enough and easier to add/remove paths in the future.

Comment (jawshooah) [inline developer/bin/sync_templates_and_ci]: Cool, done in 18422a7.

**Pull request merged**